exam_date/stored_date.py

```diff
@@ -36,38 +37,48 @@ def get_assign_submitted_date(self) -> str:
             raise e
         try:
-            path_to_persisted_file: str = self.path + '/'+self.file_name
-            self.__log.info(f"file to opened {path_to_persisted_file}")
+            self.__log.info(f"file to opened {self.path_to_persisted_file}")
 
-            if not Path(path_to_persisted_file).exists():
+            if not Path(self.path_to_persisted_file).exists():
                 self.__log.info(f"The file {self.file_name} doesn't exist this happen during the very first run")
-                latest_score_timestamp = self._create_write_persisted_file(path_to_persisted_file)
+                latest_score_timestamp = self._create_write_persisted_file()
                 return latest_score_timestamp
-            return self._read_persisted_file(path_to_persisted_file)
+            return self.read_persisted_file()
 
         except (OSError, IOError, Exception) as e:
             self.__log.error(f"error while reading the file {self.file_name} due to {e}")
             raise e
 
-    def _create_write_persisted_file(self, path_to_persisted_file: str) -> str:
+    def _create_write_persisted_file(self) -> str:
         try:
-            with open(path_to_persisted_file, 'w+') as f:
+            with open(self.path_to_persisted_file, 'w+') as f:
                 latest_score_timestamp = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
                 self.__log.info(f"""writing to file {self.file_name} the latest test taken date in
                                 UTC with current timestamp {latest_score_timestamp}""")
                 f.write(latest_score_timestamp)
                 return latest_score_timestamp
         except (OSError, IOError, Exception) as e:
-            self.__log.error(f"failed do read the file {path_to_persisted_file} due to {e}")
+            self.__log.error(f"failed do read the file {self.path_to_persisted_file} due to {e}")
             raise e
 
-    def _read_persisted_file(self, path_to_persisted_file: str) -> str:
+    def read_persisted_file(self) -> str:
         try:
-            with open(path_to_persisted_file) as f:
+            with open(self.path_to_persisted_file) as f:
                 latest_exam_date: str = f.read()
                 self.__log.info(f"""reading from to file {self.file_name} the latest test taken date in
                                 UTC with current timestamp {latest_exam_date}""")
                 return latest_exam_date.strip()
         except (OSError, IOError, Exception) as e:
-            self.__log.error(f"failed do read the file {path_to_persisted_file} due to {e}")
+            self.__log.error(f"failed do read the file {self.path_to_persisted_file} due to {e}")
+            raise e
+
+    def store_next_query_date(self, next_query_date):
+        try:
+            with open(self.path_to_persisted_file, 'w+') as f:
+                self.__log.info(f"""writing to file {self.file_name} the next query date/latest test taken date in
```

Comment:
log self.path_to_persisted_file instead of self.file_name for more information?

Reply:
done